File: ViewFinderAnywhere/src/main/java/com/fezrestia/android/util/log/Log.java
package com.fezrestia.android.util.log;

import android.os.SystemClock;

public class Log {
    // All area total log trigger.
    public static final boolean IS_DEBUG = false;

    /**
     * Debug log.
     *
     * @param tag
     * @param event
     */
    public static void logDebug(String tag, String event) {
        log("DEBUG", tag, event);
    }

    /**
     * Error log.
     *
     * @param tag
     * @param event
     */
    public static void logError(String tag, String event) {
        log("ERROR", tag, event);
    }

    private static void log(String globalTag, String localTag, String event) {
        StringBuilder builder = new StringBuilder().append("[").append(globalTag).append("] ")
                .append("[TIME = ").append(SystemClock.uptimeMillis()).append("] ")
                .append("[").append(localTag).append("]")
                .append("[").append(Thread.currentThread().getName()).append("] ")
                .append(": ").append(event);
        android.util.Log.e("TraceLog", builder.toString());
    }
}

File: ViewFinderAnywhere/src/main/java/com/fezrestia/android/lib/util/log/Log.kt
package com.fezrestia.android.lib.util.log

import android.os.SystemClock

// Total log level.
const val IS_DEBUG = false

/**
 * Debug log.
 *
 * @param tag Log tag.
 * @param event Log event.
 */
fun logD(tag: String, event: String) {
    log("DEBUG", tag, event)
}

/**
 * Error log.
 *
 * @param tag Log tag.
 * @param event Log event.
 */
fun logE(tag: String, event: String) {
    log("ERROR", tag, event)
}

private fun log(globalTag: String, localTag: String, event: String) {
    val clock = SystemClock.uptimeMillis()
    val thread = Thread.currentThread().name

    val msg = "$globalTag/ CLK=$clock TH=$thread | $localTag : $event"

    android.util.Log.e("TraceLog", msg)
}
